Add GitHub Actions workflow for building CPU Aarch64 Linux wheels

chizkiyahu opened the pull request:
### Summary

Fixes #13044

Release notes: build

Chizkiyahu commented:
@pytorchbot label "release notes: build"

Chizkiyahu commented:
@mergennachin 
Thanks for you time and help 
I build `tokenizers` on local and it works 
I open PR https://github.com/meta-pytorch/tokenizers/pull/125
please approve the workflow run

Chizkiyahu commented:
@mergennachin
thanks for the help 

I found the problem 
is missing some include https://github.com/pytorch/executorch/pull/13946/commits/bdbf309ed999d58bf682f78bcde815b35a3cb230

please approve awaiting workflows

Chizkiyahu commented:
@mergennachin 
thanks for the help
is look like that the problem is that the arm runners don't have `libatomic` install 
`libatomic`  is needed for build tokenizers

there is multiple options - what is the right way to do it 
1. to build in docker 
2. to ask pytorch/test-infra/ to install it 
3. add `sudo dnf install -y libatomic` to the pre-build script ?

or another way ?

mergennachin commented:
Asking pytorch/test-infra is probably the long term solution, but let's just do directly in the pre-build script
Something like this: 

```
  # In .ci/scripts/wheel/pre_build_script.sh
  if [[ "$(uname -m)" == "aarch64" ]]; then
      sudo dnf install -y libatomic
  fi
```

chizkiyahu committed:
Add GitHub Actions workflow for building Aarch64 Linux wheels

Chizkiyahu commented:
**Update**

1. I found a build problem with Google’s `sentencepiece` on the CI ARM machine. I wasn’t able to reproduce it on my local machine, so I cannot open an issue to `sentencepiece`.  
   The problem is resolved by [this patch](https://github.com/google/sentencepiece/commit/cf8f93b66972931504383dafcb463ed260a4a992), which fixes the linking error.  
   I updated `.ci/scripts/wheel/pre_build_script.sh` to apply this patch.  


2. Apple’s `coremltools`  does not support aarch64 https://github.com/apple/coremltools/issues/1254 and fails at runtime. I’ve updated the tests to exclude it accordingly.
